src/lifecycle.js

```diff
@@ -7,7 +7,8 @@
 const MAPPING = {
   'willStart'       : 'getInitialState',
   'willSerialize'   : 'serialize',
-  'willDeserialize' : 'deserialize'
+  'willDeserialize' : 'deserialize',
+  'willReset'       : '__AppWillReset'
```

Comment:
Sure. Let's expose it. I'll be sure to write up a pretty clear warning in the changelog